src/dc_msg.rs

```diff
@@ -1627,24 +1627,131 @@ mod tests {
         unsafe {
             let mut type_0: libc::c_int = 0;
             let mut mime_0: *mut libc::c_char = 0 as *mut libc::c_char;
+
             dc_msg_guess_msgtype_from_suffix(
                 b"foo/bar-sth.mp3\x00" as *const u8 as *const libc::c_char,
-                0 as *mut libc::c_int,
+                &mut type_0,
                 &mut mime_0,
```

Comment:
Lets pass in `type_0` and `mime_0` on all calls and check both values instead of separate calls for each.